ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -131,6 +140,14 @@ void handleStatement(
     return Optional.ofNullable(statusStore.get(statementId));
   }
 
+  public void putStatus(final CommandId commandId, final Optional<CommandStatusFuture> statusFuture,
+                        final CommandStatus status) {
+    statusStore.put(commandId, status);
+    if (statusFuture.isPresent()) {
```

Comment:
Why is this updating the state of the future, but not completing it?

Reply:
The idea here is to provide an object that StatementExecutor can use to communicate command processing status updates to KsqlResource, and that KsqlResource can block on to be notified by StatementExecutor when its done with a command. I think maybe keeping Future in the name is making it confusing, since it actually goes through multiple states, but we only want to mark it "complete" once StatementExecutor is finished. We could also decouple tracking the shared state object and wait/notify stuff.

Also, I'm not really sure what the point of the intermediate states is. If we took them out we could simplify this stuff a bit. I didn't go that route because:
    1 - I'm not totally sure its not used somewhere.
    2 - That changes the API and I wanted to keep this PR as concise as possible since its fixing a test stability problem.

Reply:
Maybe consider having a new object then, which contains the intermediate state and a future for final completion and final state?  Consider using `CompletableFuture` for the future.